rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -415,25 +423,20 @@ impl<'me, K: ArrowReadableKey<'me> + Into<KeyWrapper>, V: ArrowReadableValue<'me
             // We do not dispatch if block is present in the block manager's cache
             // but not present in the reader's cache (i.e. loaded_blocks). The
             // next read for this block using this reader instance will populate it.
-            if !self.block_manager.cached(block_id).await
-                && !self.loaded_blocks.lock().contains_key(block_id)
-            {
-                futures.push(self.get_block(*block_id));
-            }
+            // if !self.block_manager.cached(block_id).await
```

Comment:
@sanketkedia wondering if you have any insights here
`self.block_manager.cached()` seems to always return true for the blocks I try to prefetch, even when I temporarily disable the write-through cache by commenting out https://github.com/chroma-core/chroma/blob/931a5f2ee774bae040e227102e987c72b2009204/rust/blockstore/src/arrow/provider.rs#L269
but the blocks are not actually in the cache, because I can see in the resulting trace that they're read from storage during `.set()` calls
when this is commented, it works as I expect